lib/util/promises.js

```diff
@@ -0,0 +1,5 @@
+function availablePromiseMethods() {
+  return Promise.prototype.finally ? ['catch', 'finally'] : ['catch'];
```

Comment:
Hmm... based upon the Knex documentation, it looks like the `.finally(..)` method is intentionally being made available to older versions of Node.  (Specifically: release notes for version `0.20.2`).  So, perhaps we should eliminate the conditional logic here and just _always_ provide the `.finally(..)` method?  (Of course, the meta-programming might interfere with this.  So... perhaps we should avoid meta-programming for generating these methods?)

Reply:
0.20.2 is about typings only, as I can see

Reply:
These release notes for `0.20.2` on  http://knexjs.org/  mention this:

> Fix regression in older version of node when Promise#finally was not available #3507

(Perhaps this line was omitted from another copy of the release notes?)

Reply:
I checked #3507 pr